src/makeCalculator.js

```diff
@@ -37,7 +37,40 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(num) {
+      this.result += num;
+    },
+
+    subtract(num) {
+      this.result -= num;
+    },
+
+    multiply(num) {
+      this.result *= num;
+    },
+
+    divide(num) {
+      this.result /= num;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, num) {
+      callback.call(this, num);
+
+      return this;
+    },
```

Comment:
it won't work if I use some artificial callback, like
`calculator.operate((a, b) => (a + b), 42)`

let's do like this
```suggestion
    operate(callback, num) {
      this.result = callback.call(this, this.result, num);

      return this;
    },
```